.circleci/config.yml

```diff
@@ -35,10 +35,45 @@ commands:
             - run:
                   name: prepare test git user
                   command: git config --global user.email "circleci@example.com" && git config --global user.name "CircleCi Build"
+    run-regressions:
+        parameters:
+            force_update_regression_cache:
+                type: boolean
+                default: false
+            regression_color:
+                type: string
+            regression_scale:
+                type: string
+        steps:
+            - restore_cache:
+                  keys:
+                      - v1-golden-images-<< parameters.regression_color >>-<< parameters.regression_scale >>-b9e157e40b4274683707eccf7f8d400ed4c719bc
+                      - v1-golden-images-master-<< parameters.regression_color >>-<< parameters.regression_scale >>-
+            - run: yarn test:visual:ci --color=<< parameters.regression_color >> --scale=<< parameters.regression_scale >> || << parameters.force_update_regression_cache >>
+            - run: cp -RT test/visual/screenshots-current/ci test/visual/screenshots-baseline/ci
+            - when:
```

Comment:
would there be a significant downside to always pushing up this cache so we don't need to explicitly create a build before we can update the requested cache?

Reply:
actually, could we just change this just always be `on_fail` and the build branch cache to be when `on_success`?

https://circleci.com/docs/2.0/configuration-reference/#the-when-attribute
looks like they already have the when hooks that we need, then we can get rid of the `force_update_regression_cache` parameter entirely and simplify the update process to just grab the cache entry generated from failures (instead of needing to request another build with specific parameters).

Reply:
I'll look into this again. I'd used a couple of different options here attempting to leverage `on_fail` but I was probably doing something wrong. Particularly, I added the `{{ .Revision }}` gate late in the process, which _should_ be able to get us to a two-step, rather than three-step process here.

Reply:
👍 not a big deal if we can't get it down to a two-step process, happy to approve this if we can't get the `on_fail` approach working

Reply:
We can chain the `on_fail` which seems like a good way to take ou the need for `force_update_regression_cache`. Updated.